contracts/BSCValidatorSet.sol

```diff
@@ -344,6 +352,46 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     return CODE_OK;
   }
 
+  function shuffle(address[] memory validators, uint256 epochNumber, uint startIdx, uint offset, uint limit, uint modNumber) internal pure {
+    for (uint i = 0; i<limit; i++) {
+      uint random = uint(keccak256(abi.encodePacked(epochNumber, startIdx+i))) % modNumber;
+      if ( (startIdx+i) != (offset+random) ) {
+        address tmp = validators[startIdx+i];
+        validators[startIdx+i] = validators[offset+random];
+        validators[offset+random] = tmp;
+      }
+    }
+  }
+
+  function getMiningValidators(uint256 epochNumber) public view returns(address[] memory) {
```

Comment:
I think we can do not pass `epochNumber` as param, but calculate by `currentBlockHeight % EPOCH`, `EPOCH=200`.

Reply:
We wont change `EPOCH ` in near future, even we change it, it is a big hardfork.

Reply:
updated